src/components/Identity/index.js

```diff
@@ -12,28 +12,76 @@ governing permissions and limitations under the License.
 
 import { defer } from "../../utils";
 import processIdSyncs from "./processIdSyncs";
+import createEvent from "../DataCollector/createEvent";
+import {
+  serializeCustomerIds,
+  createHashFromString,
+  validateCustomerIds,
+  normalizeCustomerIds
+} from "./util";
 
-const ECID_NAMESPACE = "ECID";
+import { COOKIE_NAMES } from "./constants";
+
+const { CUSTOMER_ID_HASH, EXPERIENCE_CLOUD_ID } = COOKIE_NAMES;
 
 const addIdsContext = (payload, ecid) => {
   // TODO: Add customer ids.
-  payload.addIdentity(ECID_NAMESPACE, {
+  payload.addIdentity(EXPERIENCE_CLOUD_ID, {
     id: ecid
   });
 };
 
 const createIdentity = ({ config, logger, cookie }) => {
   // We avoid reading the ECID from the cookie right away, because we
   // need to wait for the user to opt in first.
-  const getEcid = () => cookie.get(ECID_NAMESPACE);
+  const getEcid = () => cookie.get(EXPERIENCE_CLOUD_ID);
   let optIn;
   let responseRequested = false;
   let deferredForEcid;
+  let network;
+  let lifecycle;
+  const customerIds = {};
+  const getCustomerIds = () => customerIds;
+
+  const makeServerCall = event => {
+    const payload = network.createPayload();
+    payload.addEvent(event);
+    const responsePromise = Promise.resolve()
+      .then(() => {
+        return lifecycle.onBeforeDataCollection(payload, responsePromise);
+      })
+      .then(() => {
+        return network.sendRequest(payload, payload.expectsResponse);
+      });
+
+    return responsePromise;
+  };
+
+  const setCustomerIds = ids => {
+    validateCustomerIds(ids);
+    const event = createEvent();
+    Object.keys(ids).forEach(key => {
+      customerIds[key] = ids[key];
+    });
+    const normalizedCustomerIds = normalizeCustomerIds(customerIds);
+    const customerIdsHash = createHashFromString(
+      serializeCustomerIds(normalizedCustomerIds)
+    );
+    const hasSynced = customerIdsHash === cookie.get(CUSTOMER_ID_HASH);
+    event.mergeMeta({ identity: { customerIds, hasSynced } });
```

Comment:
This is probably a dumb question because I don't really understand customer ID syncing, but why do we still send the customer IDs to the server if `hasSynced` is true?